docs/notebooks/analysis/nullpoint.ipynb

```diff
@@ -0,0 +1,235 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2bd7ab2d-427f-4d85-9d58-352aeffc0b52",
+   "metadata": {},
+   "source": [
+    "## Classifying a null point\n",
+    "Classification is done by analysing the jacobian matrix calculated at the location of null point. The full method is explained in the paper by Hynes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bdc1d3a6-35f3-4b6c-ac59-f81596db18ab",
+   "metadata": {},
+   "source": [
+    "# Running through examples\n",
+    "We will now run thorugh an example for each of the two null point finding functions to see how to properly utilize Null Point Finder. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ebcced41-d08a-405e-9f7e-df89369bca9f",
+   "metadata": {},
+   "source": [
+    "# Uniform regular grid with a model function\n",
+    "First, let's define our modeling function for the magnetic field. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "32d8553d-f114-4bc1-80d4-192070c06a01",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def magnetic_field(x, y, z):\n",
+    "    return [(y - 1.5) * (z - 1.5), (x - 1.5) * (z - 1.5), (x - 1.5) * (y - 1.5)]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "035d413d-767c-499a-8424-cd5bd96dd375",
+   "metadata": {},
+   "source": [
+    "It is easy to see that the vector field defined above, have a total of eight null points, located at $(\\pm 1.5, \\pm 1.5, \\pm 1.5)$ Now we will use `uniform_null_point_find()` to locate the null point with all positive components. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "f3edf5bb-ef1f-4622-860b-9eacc95a20ed",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[[1.49988163]\n",
+      " [1.49988163]\n",
+      " [1.49988163]]\n",
+      "Proper radial null\n"
+     ]
+    }
+   ],
+   "source": [
+    "nullpoint_args = {\n",
+    "    \"x_range\": [1, 2],\n",
+    "    \"y_range\": [1, 2],\n",
+    "    \"z_range\": [1, 2],\n",
+    "    \"precision\": [0.03, 0.03, 0.03],\n",
+    "    \"func\": magnetic_field,\n",
+    "}\n",
+    "npoints = nullpoint.uniform_nullpoint_find(**nullpoint_args)\n",
+    "print(npoints[0].loc)\n",
+    "print(npoints[0].type)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "40b3c0c5-6155-45ae-b31e-6796655db09d",
+   "metadata": {},
+   "source": [
+    "As we can see `uniform_null_point_find()` correctly identifies the location of the null point in addition to its type, which is a Proper Radial Null.  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5b30dd51-d457-4a4b-b9d8-0471f13abb7e",
+   "metadata": {},
+   "source": [
+    "# Arbitary Regular Grid\n",
+    "Now we will run through an example where the field components have to be diectly provided by the user since an adequate modeling function is not given. "
```

Comment:
```suggestion
    "Now we will run through an example where the field components have to be directly provided by the user since an adequate modeling function is not given. "
```